### PR TITLE
Smitha Hosmani

### PR DESCRIPTION
**Pull request by hsmitha26**

### CODE REVIEW

File: lib/market.rb [hunk 1]
```diff
@@ -1,3 +1,47 @@
...
+    @vendors << vendor_object
+  end
+
+  def vendor_names
+    @vendors.map {|vendor| vendor.name}
+  end
+
+  def vendors_that_sell(item_name)
+    @vendors.find_all do |vendor_object|
+      vendor_object if vendor_object.inventory.has_key?(item_name) == true
```
Comment (BrianZanti) [inline lib/market.rb]: Great job getting this to work! A refactor you could make... because `find_all` is expecting a boolean, you can remove the if logic:

```ruby
@vendors.find_all do |vendor_object|
   vendor_object.inventory.has_key?(item_name)
end
```

File: lib/market.rb [hunk 2]
```diff
@@ -1,3 +1,47 @@
...
+    end
+  end
+
+  def sorted_item_list
+    items_array = @vendors.collect do |vendor_object|
+      vendor_object.inventory.keys
+    end.flatten
+    return items_array.uniq.sort
```
Comment (BrianZanti) [inline lib/market.rb]: Great job!

File: lib/market.rb [hunk 3]
```diff
@@ -1,3 +1,47 @@
...
+    items_array = @vendors.collect do |vendor_object|
+      vendor_object.inventory
+    end
+
+    inventory_hash = {}
+    items_array.map do |item_hash|
+      item_hash.each do |key, value|
+        inventory_hash.store(key, value)
+      end
+      return inventory_hash
```
Comment (BrianZanti) [inline lib/market.rb]: The `return` keyword returns from the method, so this will cause this method to end on the first iteration of the map. You want to move this line after the map.

File: lib/market.rb [hunk 4]
```diff
@@ -1,3 +1,47 @@
...
+
+  def total_inventory
+    items_array = @vendors.collect do |vendor_object|
+      vendor_object.inventory
+    end
+
+    inventory_hash = {}
+    items_array.map do |item_hash|
+      item_hash.each do |key, value|
+        inventory_hash.store(key, value)
```
Comment (BrianZanti) [inline lib/market.rb]: Instead of `store`, what you want to do is *add* the values together:

```ruby
inventory_hash[key] += value
```

File: lib/market.rb [hunk 5]
```diff
@@ -1,3 +1,47 @@
...
+  end
+
+  def sorted_item_list
+    items_array = @vendors.collect do |vendor_object|
+      vendor_object.inventory.keys
+    end.flatten
+    return items_array.uniq.sort
+  end
+
+  def total_inventory
```
Comment (BrianZanti) [inline lib/market.rb]: Great work on this method! You were sooooo close. I think with a bit more time you would've gotten it.

File: lib/vendor.rb [hunk 6]
```diff
@@ -4,7 +4,14 @@ class Vendor
 
   def initialize(name)
     @name = name
-    @inventory = # you decide what goes here...
+    @inventory = Hash.new{|hash, key| hash[key] = 0}
```
Comment (BrianZanti) [inline lib/vendor.rb]: When you're setting an integer default value, you don't have to do this fancy passing a block thing. That's only when you want your default values to be arrays or hashes (or other objects you create if you want to get real crazy). You could change this to `Hash.new(0)`. With that said, really great job recognizing that you can use a default value for this problem 👍

File: test/market_test.rb [hunk 7]
```diff
@@ -2,7 +2,117 @@
...
+    @market.add_vendor(@vendor_1)
+    @market.add_vendor(@vendor_2)
+    @market.add_vendor(@vendor_3)
+
+    expected = [@vendor_1, @vendor_3]
+
+    assert_equal expected, @market.vendors_that_sell("Peaches")
+    assert_equal [@vendor_2], @market.vendors_that_sell("Banana Nice Cream")
+    assert_equal [],
+    @market.vendors_that_sell("Oranges")
```
Comment (BrianZanti) [inline test/market_test.rb]: This is a great edge case test!

File: test/market_test.rb [hunk 8]
```diff
@@ -2,7 +2,117 @@
 require 'minitest/pride'
 require './lib/market'
 require './lib/vendor'
+require 'pry'
 
 class MarketTest < Minitest::Test
```
Comment (BrianZanti) [inline test/market_test.rb]: Overall, your tests look great 👍

**Comment by BrianZanti** [comment]
Really great job Smitha! You were on the cusp of completing iteration 3, which is a good place to be heading into the final.
